docs/book/src/developer/core/logging.md

```diff
@@ -108,7 +115,13 @@ key value pairs (in order of importance):
   the action log and the corresponding error log; While logging before the action, log verbs should use the -ing form.
 - Ideally log messages should surface a different level of detail according to the target log level (see [log levels](#log-levels)
   for more details).
-- If Kubernetes resource name is used in log messages, it should be used as is, For example `Reconciling DockerMachineTemplate`
+- If Kubernetes resource name is used in log messages, it MUST be used as is, For example `Reconciling DockerMachineTemplate`
+- If an API field name is used in log messages, the entire path MUST be used and field names MUST capitalized like in the 
+  API (not as in the golang type). For example `Waiting for spec.providerID to be set`
+- If a log messages about a controlled or a referenced object, e.g. Machine controller performing an action on MachineSet,
+  the message MUST contain the Kind of the controlled/referenced object and its name, for example `Created MachineSet foo-bar`
+  - If the controlled/referenced object is in another namespace, use namespace/name instead of name
```

Comment:
Sure to repeat the object name, which should already be a structured field? I'm not against this repetition to make it more readable for _humans_ – just asking if this was considered.

The namespace should be in the structured field automatically (via `klog.KRef`). If we want the human log message to only contain the namespace if it's outside the "default", then I guess this means a utility function which does that?